src/xrpld/app/misc/CredentialHelpers.cpp

```diff
@@ -186,15 +185,62 @@ valid(PreclaimContext const& ctx, AccountID const& src)
 }
 
 TER
-authorized(ApplyContext const& ctx, AccountID const& dst)
+validDomain(ReadView const& view, uint256 domainID, AccountID const& subject)
+{
+    // Note, permissioned domain objects can be deleted at any time
+    auto const slePD = view.read(keylet::permissionedDomain(domainID));
+    if (!slePD)
+        return tecINVALID_DOMAIN;
+    else if (!slePD->isFieldPresent(sfAcceptedCredentials))
+        return tefINTERNAL;
+
+    auto const closeTime = view.info().parentCloseTime;
+    bool foundExpired = false;
+    for (auto const& h : slePD->getFieldArray(sfAcceptedCredentials))
+    {
+        if (!h.isFieldPresent(sfIssuer) || !h.isFieldPresent(sfCredentialType))
+            return tefINTERNAL;
+
+        auto const issuer = h.getAccountID(sfIssuer);
```

Comment:
just a fyi, the domain issuer is considered to be in the domain themself, at least in permissioned dex's context.